docs/reference/security-hardening/webhook-mtls.md

```diff
@@ -0,0 +1,188 @@
+---
+sidebar_label: Webhook Mutual TLS
+title: Secure webhooks with mutual TLS
+description: Harden the webhook configuration.
+keywords: [kubewarden, kubernetes, security]
+doc-persona: [kubewarden-operator, kubewarden-integrator]
+doc-type: [reference]
+doc-topic: [reference, security]
+---
+
+<head>
+  <link rel="canonical" href="https://docs.kubewarden.io/reference/security-hardening/webhook-mtls"/>
+</head>
+
+The Kubewarden stack uses different webhooks to enforce policies on the Kubernetes cluster. Each `PolicyServer` instance exposes a webhook
+that the Kubernetes API server calls to validate and mutate resources. Moreover, the `kubewarden-controller` exposes webhooks to validate and mutate
+the custom resources provided by the Kubewarden project.
+
+To decrease their attack surface, access to these webhooks should be limited to the only valid callers they have: the Kubernetes API server
+and the [audit scanner](../../explanations/audit-scanner/) component.
+This can be done by using network policies and authentication independently or in conjunction with each other to harden the webhooks against attacks.
+
+## Block External Traffic Using Network Policies
+
+The webhooks are only expected to accept requests from the Kubernetes API server and from the audit scanner component. By default, however, the webhooks
+can accept traffic from any source.
+If you are using a CNI that supports Network Policies, you can create a policy that blocks traffic that doesn't originate from the API server.
+
+The built-in NetworkPolicy resource in Kubernetes can't block or admit traffic from the cluster hosts, and the `kube-apiserver` process is always running on the host network.
+Therefore, you must use the advanced network policy resources from the CNI in use. Examples for Calico and Cilium follow. Consult the documentation for your CNI for more details.
+
+### Calico
+
+Use the NetworkPolicy resource in the `crd.projectcalico.org/v1` API group, define a network policy like the following one:
+
+```yaml
+apiVersion: crd.projectcalico.org/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-k8s-and-audit-scanner
+  namespace: kubewarden # namespace where the kubewarden stack is deployed
+spec:
+  selector: 'app.kubernetes.io/name == "kubewarden-controller" || has(kubewarden/policy-server)'
```

Comment:
Should we update to the new labels already, with a `managed-by`?

Reply:
Yeah, I agree. As we are about to merge the [PR](https://github.com/kubewarden/kubewarden-controller/pull/1037) adding the recommended policies, I think we can hold this PR until we merge that one.